CodeEdit/Breadcrumbs/BreadcrumbsView.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  BreadcrumbsView.swift
+//  CodeEdit
+//
+//  Created by Lukas Pistrol on 17.03.22.
+//
+
+import SwiftUI
+import WorkspaceClient
+
+struct BreadcrumbsView: View {
+
+	@ObservedObject var workspace: WorkspaceDocument
+	var file: WorkspaceClient.FileItem
+
+	@State private var projectName: String = ""
+	@State private var folders: [String] = []
+	@State private var fileName: String = ""
+	@State private var fileImage: String = "doc"
+
+	init(_ file: WorkspaceClient.FileItem, workspace: WorkspaceDocument) {
+		self.file = file
+		self.workspace = workspace
+	}
+
+    var body: some View {
+		ZStack(alignment: .leading) {
+			Rectangle()
+				.foregroundStyle(Color(nsColor: .controlBackgroundColor))
+			HStack {
+				BreadcrumbsComponent(projectName, systemImage: "square.dashed.inset.filled", color: .accentColor)
+				chevron
+				ForEach(folders, id:\.self) { folder in
+					BreadcrumbsComponent(folder, systemImage: "folder.fill")
+					chevron
+				}
+				BreadcrumbsComponent(fileName, systemImage: fileImage, color: .accentColor)
+			}
+			.padding(.leading, 12)
+		}
+		.frame(height: 29)
+		.overlay(alignment: .bottom) {
+			Divider()
+		}
+		.onAppear {
+			fileInfo()
+		}
+    }
+
+	private var chevron: some View {
+		Image(systemName: "chevron.compact.right")
+			.foregroundStyle(.secondary)
+			.imageScale(.large)
+	}
+
+	private func fileInfo() {
+		guard let projName = workspace.workspaceClient?.folderURL()?.lastPathComponent else { return }
+		var components = file.url.pathComponents.split(separator: projName).last!
```

Comment:
we should try to avoid force casts... I think we can just this inside the guard so we don't have to force it

Reply:
good point 😄

Reply:
like

```swift
guard let projName = workspace.workspaceClient?.folderURL()?.lastPathComponent,
      let components = file.url.pathComponents.split(separator: projName).last else { return }
```

Reply:
yeah this should work great!

Reply:
`let components` will be changed to `var components` since it needs to be mutable.